packages/webapp/src/elections/components/ongoing-election-components/ongoing-round-segment.tsx

```diff
@@ -1,79 +1,160 @@
-import { ActiveStateConfigType, ElectionStatus } from "elections/interfaces";
+import { Button, Container, Expander, Heading, Loader, Text } from "_app/ui";
 import { MembersGrid } from "members";
 import { MemberData } from "members/interfaces";
-import { setVote } from "../../transactions";
+import { ElectionParticipantChip, VotePieMer } from "elections";
+import {
+    ActiveStateConfigType,
+    ElectionStatus,
+    RoundStage,
+} from "elections/interfaces";
 
 import Consensometer from "./consensometer";
-import PasswordPromptModal from "./password-prompt-modal";
+import ErrorLoadingElection from "./error-loading-election";
 import RoundHeader from "./round-header";
+import VotingRoundParticipants from "./voting-round-participants";
+import { setVote } from "../../transactions";
 
 export interface RoundSegmentProps {
     electionState: string;
     roundIndex: number;
-    roundEndTime: string;
+    roundStartTime: Dayjs;
+    roundEndTime: Dayjs;
+    roundDurationMs: number;
     electionConfig?: ActiveStateConfigType;
 }
 
 // TODO: Much of the building up of the data shouldn't be done in the UI layer. What do we want the API to provide? What data does this UI really need? We could even define a new OngoingElection type to provide to this UI.
 export const OngoingRoundSegment = ({
     electionState,
     roundIndex,
+    roundStartTime,
     roundEndTime,
+    roundDurationMs,
     electionConfig,
 }: RoundSegmentProps) => {
     const queryClient = useQueryClient();
 
+    // duration of time periods before and after election meeting call
+    // stages: meeting prep -> meeting -> post-meeting finalization -> round end
+    const meetingBreakDurationMs = (roundDurationMs - meetingDurationMs) / 2;
+
+    const now = dayjs();
+
+    const meetingStartTime = roundStartTime.add(meetingBreakDurationMs);
+    const postMeetingStartTime = meetingStartTime.add(meetingDurationMs);
+
+    let currentStage = RoundStage.PreMeeting;
+    let timeRemainingToNextStageMs: number | null = meetingStartTime.diff(now);
+    let currentStageEndTime: Dayjs = meetingStartTime;
+
+    if (now.isAfter(roundEndTime)) {
+        currentStage = RoundStage.Complete;
+        timeRemainingToNextStageMs = null;
+        currentStageEndTime = roundEndTime;
+    } else if (now.isAfter(postMeetingStartTime)) {
+        currentStage = RoundStage.PostMeeting;
+        timeRemainingToNextStageMs = roundEndTime.diff(now);
+        currentStageEndTime = roundEndTime;
+    } else if (now.isAfter(meetingStartTime)) {
+        currentStage = RoundStage.Meeting;
+        timeRemainingToNextStageMs = postMeetingStartTime.diff(now);
+        currentStageEndTime = postMeetingStartTime;
+    }
+
+    const [stage, setStage] = useState<RoundStage>(currentStage);
+
+    const isVotingOpen = [RoundStage.Meeting, RoundStage.PostMeeting].includes(
+        stage
+    );
+
+    useTimeout(() => {
+        setStage(stage + 1);
+    }, timeRemainingToNextStageMs);
```

Comment:
A lot of this is also a good candidate for factoring out. I could see a hook...or at least a util coming from this.